app/controllers/little_shop_app.rb

```diff
@@ -119,5 +119,20 @@ class LittleShopApp < Sinatra::Base
     erb :"merchants/dashboard"
   end
 
+  get "/search" do
+    if params[:q] == ""
+      @query = nil
+    else
+      @query = [Category.where(name: params[:q]),
+                Merchant.where(name: params[:q]),
+                Item.where(title: params[:q])]
+
+      @categories = @query[0]
+      @merchants = @query[1]
+      @items = @query[2]
+    end
+    erb :"/search"
+  end
 
```

Comment:
this is great! can we look into redirects as well? i.e. if a merchant or item name/id is matched it sends us straight to the show page, and if not then we are lead to the search page?